tests/baselines/reference/user/bluebird.log

```diff
@@ -3,90 +3,23 @@ Standard output:
 node_modules/bluebird/js/release/assert.js(11,30): error TS2339: Property 'constructor$' does not exist on type 'Error'.
 node_modules/bluebird/js/release/bluebird.js(5,15): error TS2367: This condition will always return 'false' since the types 'PromiseConstructor' and 'typeof Promise' have no overlap.
 node_modules/bluebird/js/release/bluebird.js(10,10): error TS2339: Property 'noConflict' does not exist on type 'typeof Promise'.
-node_modules/bluebird/js/release/call_get.js(11,24): error TS2339: Property 'canEvaluate' does not exist on type 'typeof ret'.
```

Comment:
fails to import via `require`, not sure why yet

Reply:
I remember this now:

```js
// @filename: first.js
var ret = {
  global: this,
  isClass: true
}
module.exports = ret
// @filename: test.js
var o = require('./first')
o./**/
```

`this` is a reference to the module, which is then exported, so this reference is circular. My simplification of type reference resolution stopped eagerly resolving this type, but then my fix started again. It would be nicer to not eagerly resolve, but not required. This is an OK break -- and goes back to the way it was in 3.6.